Classes/ContentObject/Relation.php

```diff
@@ -306,42 +316,51 @@ protected function resolveRelatedValue(
         if (Util::getLanguageUid() > 0 && !empty($foreignTableName)) {
             $relatedRecord = $this->frontendOverlayService->getOverlay($foreignTableName, $relatedRecord);
         }
+        $values = [];
```

Comment:
My use case did not cover this part of the code. That's why I was unable to find a test case for that.
However since this also uses the foreignTableLabelField setting it would probably break when allowing multiple values. So I adjusted that as well. Maybe you can help me with a test case for this one. ;)